pvCopy: implement plugin filters

mrkraimer opened the pull request:
mplement plugin filters for copy facility.
This is modeled after the plugin filters that were introduced in epics base 3.15.

mrkraimer committed:
add Filter Plugin support for PVCopy

mrkraimer committed:
more work on pvCopy plugins; channelGet, channelPut, and monitor now work

mrkraimer committed:
deadband filter; more work on filter implementations; CreateRequest now allows top level options

mrkraimer committed:
fix issue #13 in epics-base; more work on plugins for pvCopy

mrkraimer committed:
add README.md; fix bug in CreateRequest while adding ability to have top level options

mrkraimer committed:
many minor changes to copy facility

mrkraimer committed:
fix CreateRequest bug; update doc; remove MonitorAlgorithm

duchuyctlk commented on src/org/epics/pvdata/copy/arrayPlugin/ArrayFilter.java:
```diff
@@ -0,0 +1,149 @@
+        int num = values.length;
+        if(num==1) {
+            String value = values[0];
+            try{
```
This `try-catch` block is used multiple times. Please consider extract it into a method.

mrkraimer committed:
PVAlarm,PVTimeStamp,Display,Control: only update fields that have changed

kasemir commented:
Isn't the alarm message a String, so this
```
if(current.getMessage()!=alarm.getMessage()) 
```
should be
```
if (!current.getMessage().equals(alarm.getMessage())) 
```
or in case there could be nulls,
```
if (! Objects.equals(current.getMessage(), alarm.getMessage())) 
```
?

mrkraimer commented:
Kay is correct.
It also is a problem with the changes I just made to PVAlarm, and PVDisplay.
We should discuss at todays hangout.
Kay please join the hangout.

ralphlange commented:
If I remember correctly, earlier this year we agreed to not further develop this idea until there is a user request for it.

mrkraimer committed:
Backout plugin support.
PVAlarmFactory and PVDisplayFactory: in set method fix bug for string fields
PVTimeStampFactory: In set methods fix bug for nanaSeconds.
ConvertFactory: make some changes for unsigned field types. This removes warning message from compiler.

mrkraimer commented:
I am closing this pull request and will issue a new one based on decisions made at  Oct 17, 2017 hangout
